sno/working_copy/gpkg_adapter.py

```diff
@@ -474,29 +485,111 @@ def json_to_gpkg_metadata(v2_metadata_json, table_name, reference=False):
     return result
 
 
+_KNOWN_METADATA_URIS = {
+    "GDALMultiDomainMetadata": "http://gdal.org",
+}
+
+
+def _get_uri_from_xml(xml_metadata):
+    from xml.dom.minidom import parseString
+
+    try:
+        doc = parseString(xml_metadata)
+        element = doc.documentElement
+        return (
+            _KNOWN_METADATA_URIS.get(element.tagName)
```

Comment:
I don't think we should be using GDALs metadata — is this an artifact of the OGR importer? It's basically a wrapper of actual metadata standards.

Need tests for this.

Reply:
We don't "use" GDAL metadata as such. We accept any XML metadata that is attached and store it unchanged. The OGR importer doesn't provide any metadata (as far as I know, in any case we don't use it) - it's up to us to extract the metadata. We do this using the same code we use for extracting metadata from the GPKG working copy.

The code for guessing the URI is now moved from the importer to here. I can add some tests.

Reply:
In principle yes, we accept whatever, but GDAL is a wrapper around _actual_ metadata :)

If there's no path to adding it at import time I think it's just a case of fixing the test data. I guess I just overlooked it.

Reply:
As above:
"Fixed XML for polygons test repo in particular, in a separate commit.
Weird test data is unique to polygons test repo and doesn't seem to point to any particular Sno issue."

Code for guessing URI still has no tests - I will add tests in a separate PR.